pkg/controller/nodeconfigpod/sync.go

```diff
@@ -39,6 +39,11 @@ func (ncpc *Controller) sync(ctx context.Context, key string) error {
 		return fmt.Errorf("can't list pods: %w", err)
 	}
 
+	_, hasScyllaLabel := pod.Labels[naming.ClusterNameLabel]
+	if !hasScyllaLabel {
+		return nil
```

Comment:
this should log a warning given we also pre-filter in the handlers unless there is some good path how it can land here in which case I'd still log it at least at info 4